src/braket/parametric/free_parameter.py

```diff
@@ -99,6 +105,22 @@ def to_dict(self) -> dict:
             "name": self.name,
         }
 
+    @staticmethod
+    def _validate_name(name: str) -> None:
+        if not name:
+            raise ValueError("Symbol names must be non empty")
+        if not isinstance(name, str):
+            raise TypeError("Symbol name must be a string")
+        if name.startswith("_"):
+            if name.startswith("__"):
+                raise ValueError("Symbol names must not start with two underscores '__'")
+        else:
+            unicode_category = category(name[0])
+            if not unicode_category:
+                raise ValueError("Symbol names must start with a valid character")
+            if not unicode_category.startswith("L") and unicode_category != "Nl":
```

Comment:
This breaks my use of unsightly circuits. 

```
\U0001f600
```
is no longer valid. This raises the question of whether we should exploring fuzzing certain parts of the BDK...

Reply:
This also seems to break in the case we have:

```
FreeParameter("_a")
```

Reply:
_a should still work. \U0001f600 can't be converted to OpenQASM, so we should probably disallow it. What do you think?

Reply:
Yes, I reduced the scope of the change to only disallow __